### PR TITLE
Release workflow

### PR DESCRIPTION
**Pull request by grischperl**
**Description**

Add Github Action to create a release.

Changes proposed in this pull request:

- Triggered manually without inputs
- Uses branch name `release-x.y` as input for release version `x.y.z`

**Related issue(s)**

[kyma/#18477](https://github.com/kyma-project/kyma/issues/18477)

### CODE REVIEW

File: scripts/check_tag_info.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+##############################
+# Check tags in security-scan-config.yaml
```
Comment (friedrichwilken) [inline scripts/check_tag_info.sh:4]:
```suggestion
# Check tags in sec-scanners-config.yaml
```

File: .github/workflows/create-release.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            echo "current_branch=$CURRENT_BRANCH" >> $GITHUB_ENV
+            echo "Branch name starts with 'release-'."
+          else
+            echo "Branch name does not start with 'release-'."
+            exit 1
+          fi
+      - name: Get the next release version
+        id: release-version
+        run: |
+          RELEASE_VERSION=$(./scripts/check_artifacts_existence.sh "$current_branch")
```
Comment (friedrichwilken) [inline .github/workflows/create-release.yaml:31]: you are missing the script `./scripts/check_artifacts_existence.sh` on main.

File: .github/workflows/create-release.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        run: |
+          CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+          if [[ "$CURRENT_BRANCH" == release-* ]]; then
+            echo "current_branch=$CURRENT_BRANCH" >> $GITHUB_ENV
+            echo "Branch name starts with 'release-'."
+          else
+            echo "Branch name does not start with 'release-'."
+            exit 1
+          fi
+      - name: Get the next release version
```
Comment (friedrichwilken) [inline .github/workflows/create-release.yaml:28]: for consistentcy with our other gh workflows, can you please put empty lines in between steps?
```suggestion
          fi
          
      - name: Get the next release version
```

**Review by friedrichwilken — CHANGES_REQUESTED** [review]

File: scripts/check_tag_info.sh (hunk 1 above)
Comment (grischperl) [reply scripts/check_tag_info.sh]: Done

File: .github/workflows/create-release.yaml (hunk 2 above)
Comment (grischperl) [reply .github/workflows/create-release.yaml]: Fixed

File: .github/workflows/create-release.yaml (hunk 3 above)
Comment (grischperl) [reply .github/workflows/create-release.yaml]: Done

File: scripts/check_tag_info.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+##############################
+# Check tags in sec-scan-config.yaml
```
Comment (friedrichwilken) [inline scripts/check_tag_info.sh:4]:
```suggestion
# Check tags in sec-scanners-config.yaml
```

File: .github/workflows/create-release.yaml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        id: release-version
+        run: |
+          RELEASE_VERSION=$(./scripts/get_next_release_version.sh "$current_branch")
+          echo "release_version=$RELEASE_VERSION" >> $GITHUB_OUTPUT
+
+      - name: Check image tag
+        env:
+          RELEASE_VERSION: ${{ steps.release-version.outputs.release_version }}
+        run: ./scripts/check_tag_info.sh $RELEASE_VERSION
```
Comment (friedrichwilken) [inline .github/workflows/create-release.yaml:41]: let's give this script a name that make clear, what it really does. `check-sec-scanners-config.sh` or something similar.

File: .github/workflows/create-release.yaml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+            exit 1
+          fi
+
+      - name: Get the next release version
+        id: release-version
+        run: |
+          RELEASE_VERSION=$(./scripts/get_next_release_version.sh "$current_branch")
+          echo "release_version=$RELEASE_VERSION" >> $GITHUB_OUTPUT
+
+      - name: Check image tag
```
Comment (friedrichwilken) [inline .github/workflows/create-release.yaml:38]: let's give this give this step a name that makes clear, what it really does. `check sec-scanners-config.yaml` for example.

File: scripts/create_changelog.sh [hunk 7]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
```
Comment (friedrichwilken) [inline scripts/create_changelog.sh:1]: we have an updated versions of this file with more comments. Let's use this instead: https://github.com/kyma-project/nats-manager/blob/main/.github/scripts/create_changelog.sh

File: scripts/create_draft_release.sh
Comment (friedrichwilken) [inline scripts/create_draft_release.sh:1]: we have an updated version of this files with more comments: https://github.com/kyma-project/nats-manager/blob/main/.github/scripts/create_draft_release.sh

File: .github/workflows/create-release.yaml [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+      - name: Create the draft release
+        id: create-draft
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          RELEASE_ID=$(./scripts/create_draft_release.sh $RELEASE_VERSION)
+          echo "release_id=$RELEASE_ID" >> $GITHUB_OUTPUT
+
+      - name: Add lightweight tag
```
Comment (friedrichwilken) [inline .github/workflows/create-release.yaml:70]: This step happens to trigger the prow build job and I remember that it was not so easy to see that when I looked at this the first time. Why don't we leave a nice comment here that points this out?

File: .github/workflows/create-release.yaml [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+        run: |
+          git tag $RELEASE_VERSION
+          git push origin $RELEASE_VERSION
+
+      - name: Verify job status
```
Comment (friedrichwilken) [inline .github/workflows/create-release.yaml:75]:
```suggestion
      - name: Verify build job status
```

File: scripts/verify_status.sh
Comment (friedrichwilken) [inline scripts/verify_status.sh:1]: let's give this file a name that makes clear what status it verifies. Like `verify_prow_build_job_status.sh` or something similar.

**Review by friedrichwilken — CHANGES_REQUESTED** [review]

File: .github/workflows/create-release.yaml (hunk 6 above)
Comment (grischperl) [reply .github/workflows/create-release.yaml:38]: Done

File: .github/workflows/create-release.yaml (hunk 5 above)
Comment (grischperl) [reply .github/workflows/create-release.yaml:41]: Done

File: .github/workflows/create-release.yaml (hunk 9 above)
Comment (grischperl) [reply .github/workflows/create-release.yaml:75]: Done

File: scripts/check_tag_info.sh (hunk 4 above)
Comment (grischperl) [reply scripts/check_tag_info.sh:4]: Done

File: .github/workflows/create-release.yaml (hunk 8 above)
Comment (grischperl) [reply .github/workflows/create-release.yaml]: Done

File: scripts/create_changelog.sh (hunk 7 above)
Comment (grischperl) [reply scripts/create_changelog.sh:1]: Done

File: .github/workflows/create-release.yaml [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+      - name: Checkout EPP repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Verify that the current branch has a name that starts with 'release-'
+        run: |
+          CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+          if [[ "$CURRENT_BRANCH" == release-* ]]; then
+            echo "current_branch=$CURRENT_BRANCH" >> $GITHUB_ENV
```
Comment (friedrichwilken) [inline .github/workflows/create-release.yaml:25]: Why not simply
```suggestion
            echo "$CURRENT_BRANCH" >> $GITHUB_ENV
```
and use $CURRENT_BRANCH" in the following steps?

Comment (grischperl) [reply .github/workflows/create-release.yaml:25]: As I understood it, I need a variable name to set an environment variable: https://docs.github.com/en/github-ae@latest/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

**Review by friedrichwilken — APPROVED** [review]

**Pull request merged**